pkg/network/nmstate.go

```diff
@@ -23,7 +23,7 @@ const (
 
 // renderNMState generates the manifests of NMState handler
 func renderNMState(conf *cnao.NetworkAddonsConfigSpec, manifestDir string, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
-	if conf.NMState == nil {
+	if conf.NMState == nil || clusterInfo.NmstateCRExists {
```

Comment:
shouldn't the logic be 
```
if conf.NMState == nil || (clusterInfo.NmstateCRExists && clusterInfo.NmstateOperator) {
```
?

Reply:
Totally!

Reply:
[Done](https://github.com/kubevirt/cluster-network-addons-operator/compare/08d160a48ed4b089cc654761fd1112b3bb3a3e68..35f776d9be08fce4039c5feeb2a2cfa0f8faec9e)